pyteal/ast/gitxn.py

```diff
@@ -46,8 +47,10 @@ def __teal__(self, options: "CompileOptions"):
 class GitxnaExpr(TxnaExpr):
     """An expression that accesses an inner transaction array field from an inner transaction in the last inner group."""
 
-    def __init__(self, txnIndex: int, field: TxnField, index: int) -> None:
-        super().__init__(Op.gitxna, None, "Gitxna", field, index)
+    def __init__(
+        self, txnIndex: Union[int, Expr], field: TxnField, index: Union[int, Expr]
+    ) -> None:
+        super().__init__(Op.gitxna, Op.gitxnas, "Gitxna", field, index)
```

Comment:
nit: I might be wrong but according to: https://github.com/algorand/go-algorand/tree/master/data/transactions/logic#inner-transactions, it seems `txnIndex` is still an immediate value, which can only be `int`.

What makes the difference in choice of `gitxna` and `gitxnas` is where the last `index` comes from:
- `index` is a stack value (`Expr`), then you want `gitxnas`
- `index` is an int (immediate value), then you want `gitxna`

Reply:
@ahangsu Thanks for catching and for the context.  

You're definitely right.  I misunderstood the change's implications.  Via https://github.com/algorand/pyteal/pull/193/commits/1a78b6beac9a70da56e16dbf4d5cc96c6d7e5931, I attempted to fix the mistake.

Reply:
Nice, let me take a pass again :-)